api/server/graph.cc

```diff
@@ -521,8 +521,10 @@ bool Graph::NicAdd(app::Nic *nic_) {
     }
     if (nic.packet_trace) {
         name = "sniffer-" + gn.id;
-        gn.pcap_file = fopen(("/tmp/butterfly-" + std::to_string(getpid()) +
-                              "-" + gn.id + ".pcap").c_str(), "w");
+        gn.packet_trace_path = nic.packet_trace_path;
+        gn.pcap_file = fopen((nic.packet_trace_path + "/butterfly-" +
```

Comment:
so, titi.pcap will create "titi.pcap/butterfly-PID-ID"
sorry that's just not what was asked
the goal of nic.packet_trace_path is to allow the user to chose the full path of his pcap
not the directory where the pcap file will be found

Reply:
- ```titi.pcap will create "titi.pcap/butterfly-PID-ID"``` 
    Yes if "titi.pcap" is a directory. I think the PATH mean the directory where to find something, rigth?

- You said : "the goal of nic.packet_trace_path is to allow the user to chose the full path of his pcap not the directory where the pcap file will be found" 
   So you mean we have to change the PATH and rename the pcap files if i understand